lib/elixir/lib/enum.ex

```diff
@@ -723,28 +723,30 @@ defmodule Enum do
   end
 
   def fetch(enumerable, index) when is_integer(index) and index < 0 do
-    case Enumerable.count(enumerable) do
-      {:error, _module} ->
+    module = Enumerable.impl_for!(enumerable)
+    case module.count(enumerable) do
```

Comment:
Sure, I will revert.

Could you explain where the second dispatch happens in my implementation, though? The first is during `impl_for!` call, but I can't find the second one.